src/app/util/attribute-storage.cpp

```diff
@@ -261,9 +261,43 @@ uint16_t emberAfGetDynamicIndexFromEndpoint(EndpointId id)
     return kEmberInvalidEndpointIndex;
 }
 
+const EmberAfCluster * getClusterTypeDefinition(EndpointId endpointId, ClusterId clusterId, EmberAfClusterMask mask)
+{
+    uint16_t index = emberAfIndexFromEndpointIncludingDisabledEndpoints(endpointId);
+    if (index != kEmberInvalidEndpointIndex)
+    {
+        return emberAfFindClusterInType(emAfEndpoints[index].endpointType, clusterId, mask, nullptr);
+    }
+    // not found
+    return nullptr;
+}
+
+CHIP_ERROR setupDynamicEndpointDeclaration(EmberAfEndpointType & endpointType, EndpointId templateEndpointId,
```

Comment:
This does not seem to ever return anything except CHIP_NO_ERROR (it verifies or dies). Should we change this to return instead of VerifyOrDie? Or alternatively we should make this void and document that it crashes on invalid data or internal failures.

Reply:
Thanks for pointing out. I inititially thought of handling this at runtime, so prepared for returning an error. But later I realized trying to instantiate non-existing clusters means that the app is severely broken and there's no good runtime action one could take.

So I stayed with VerifyOrDie and void and documented that.